### PR TITLE
Fix error with empty style definitions json file

### PR DESCRIPTION
**Pull request by EasterPeanut**
An artifact in our own codebase could sometimes cause empty style definitions json files. This will fail compilation of ex_css_modules because decoding an empty string is not working.

To capture this edge case, let's check if the file is empty before trying to decode OR generate the json file through a task.

### CODE REVIEW

File: lib/ex_css_modules/ex_css_modules.ex [hunk 1]
```diff
@@ -37,21 +37,40 @@ defmodule ExCSSModules do
...
+      cond do
+        File.exists?("#{filename}.json") && not empty_file?("#{filename}.json") ->
+          "#{filename}.json"
           |> File.read!()
           |> json_library().decode!()
-        end
 
-      true ->
-        %{}
+        not is_nil(build_json_task) ->
```
Comment (sn3p) [inline lib/ex_css_modules/ex_css_modules.ex:47]: Match for task/module here?

File: lib/ex_css_modules/ex_css_modules.ex [hunk 2]
```diff
@@ -37,24 +37,45 @@ defmodule ExCSSModules do
...
+        not File.exists?("#{filename}.json") ->
+          message =
+            """
+            File does not exist.
+            Error compiling ExCSSModules: Be sure to build your style definitions JSON files before compiling your application.
+            """
+
+          raise(CompileError, description: message, file: "#{filename}.json")
+
+        empty_file?(filename) ->
```
Comment (sn3p) [inline lib/ex_css_modules/ex_css_modules.ex:63]:
```suggestion
        empty_file?("#{filename}.json") ->
```

no?

File: README.md [hunk 3]
```diff
@@ -28,7 +32,18 @@ defmodule Mix.Tasks.GenerateMissingJson do
...
 ```
 
-If you don't set the flag or set it to false, the stylesheet definition JSON files are read live from the server which creates a lot of IO for each request.
+This task can be passed through the option `build_stylesheet_definitions_json_task`: 
+
+```ex
+use ExCSSModules.View,
+  namespace: MyApplicationWeb,
+  embed_stylesheet: ApplicationSettings.built_for?(:prod),
+  build_stylesheet_definitions_json_task: unquote(Mix.Tasks.BuildScopedJson),
```
Comment (EasterPeanut) [inline README.md:41]:
```suggestion
  build_stylesheet_definitions_json_task: unquote(Mix.Tasks.GenerateMissingJson),
```

**Comment by zwippie** [comment]
Tested in detroit with this branch. Added `build_stylesheet_definitions_json_task` option in web.ex. Deleted a random style.css json file. Results was a `(UndefinedFunctionError) function Mix.Tasks.GenerateMissingJson.run/1 is undefined` but I guess that task is on its way.

Nice!

**Comment by EasterPeanut** [comment]
> Tested in detroit with this branch. Added `build_stylesheet_definitions_json_task` option in web.ex. Deleted a random style.css json file. Results was a `(UndefinedFunctionError) function Mix.Tasks.GenerateMissingJson.run/1 is undefined` but I guess that task is on its way.
> 
> 
> 
> Nice! 

Yes! That will be done in https://github.com/DefactoSoftware/detroit/pull/13193 :D

**Pull request merged**